prow/cmd/crier/README.md

```diff
@@ -66,3 +66,36 @@ key to the shared cache and retry up to 5 times.
 
 You can add a reporter that implements the above interface, and add a flag to turn it on/off in crier.
 
+## Migration from plank for github report
+
+Both plank and crier will into the [github report lib](prow/github/report) when a prowjob needs to be reported,
+so as a user you only want to make one of them to report :-)
+
+Before migrating, be sure plank is setting the [PrevReportStatus field](https://github.com/kubernetes/test-infra/blob/master/prow/apis/prowjobs/v1/types.go#L403)
+by describing a finished presubmit prowjob, you should see an entry like:
+```sh
+senlu@senlu:~/work/src/k8s.io/test-infra/prow$ kubectl describe prowjob 7b4dc5c8-3afd-11e9-abfb-0a580a6c0a37
+...
+Status:
+  ...
+  Prev _ Report _ States:
+    Github - Reporter:  success
+  State:                success
+  ...
+...
+
+```
+
+You want to add a crier deployment, similar to ours [prow/cluster/crier_deployment.yaml](prow/cluster/crier_deployment.yaml),
+flags need to be specified:
+- point `config-path` and `--job-config-path` to your prow config and job configs accordingly.
+- Set `--github-worker` to be number of parallel github reporting threads you need
+- Point `--github-endpoint` to ghproxy, if you have set that for plank
+- Bind github oauth token as a secret and set `--github-token-path` if you've have that set for plank.
+
+In your plank deployment, you can
+- Remove the `--github-endpoint` flags
+- Remove the github oauth secret, and `--github-token-path` flag if set
+- Flip on `--skip-report`, so plank will skip the reporting logic
```

Comment:
What ordering is needed here?

Reply:
technically at the same time

Reply:
As in, it will race? I cannot do this atomically, so one must happen first. Which is it? Or does it not matter?

Reply:
I think crier change should go first then - report twice is probably fine and should be no-op from the ghproxy? (Also technically we can lose informer events between crier init and plank down?)

(sorry I was super backlogged on this migration with catching up the state with our Prow, and we still only report the build controller jobs via crier)

Reply:
briefly tried with https://github.com/GoogleCloudPlatform/oss-test-infra/pull/6 for github reporting which looks fine so far